src/FirebirdSql.Data.FirebirdClient/FirebirdClient/FbBatchCommand.cs

```diff
@@ -1068,6 +1068,17 @@ private async ValueTask UpdateParameterValuesAsync(Descriptor descriptor, int ba
 				{
 					batchParameter.NullFlag = 0;
 
+					if (commandParameter.Charset > 0 && Charset.TryGetByType(commandParameter.Charset, out var charset))
+					{
+						//if command parameter charset is explicitly set, and we can find it, apply it to the statement parameter
+						batchParameter.Charset = charset;
+					}
+					else if (!Connection.InnerConnection.Database.Charset.IsNoneCharset)
+					{
+						//if the database charset is not NONE (default), apply the charset to the statement parameter
+						batchParameter.Charset = Connection.InnerConnection.Database.Charset;
```

Comment:
This is probably not a good idea, because of backward compatibility. Or at least we (I) need understand the real world implications.